docs/contributing/component-js-configuration-options.md

```diff
@@ -0,0 +1,192 @@
+
+```javascript
+import { mergeConfigs } from '../../common/index.mjs'
+
+this.config = mergeConfigs(
+  defaultConfig,
+  config || {}
+)
+```
+
+We can now reference the configuration option anywhere we need to in the component's JavaScript:
+
+```javascript
+this.config.rememberExpanded
+// => true
+```
+
+It's now possible to individually initialising the component with configuration options.
+
+```html
+<script>
+  var $accordion = document.getElementById("accordion")
+  new window.GOVUKFrontend.Accordion($accordion, {
+    rememberExpanded: false
+  }).init()
+</script>
+```
+
+## Allowing options to be passed through the `initAll` function
+
+Usually, teams will not be individually initialising components. Instead, GOV.UK Frontend ships with an `initAll` function, which searches the page for instances of components and automatically initialises them.
+
+In `src/govuk/all.mjs`, update the component's `new` function to pass through a nested configuration object. The nested object should use the component's name converted to camelCase (for example, the 'Character Count' component becomes `characterCount`).
+
+```javascript
+new Accordion($accordion, config.accordion).init()
+```
+
+It's now possible to pass configuration options for your component, as well as multiple other components, via the `initAll` function.
+
+```html
+<script>
+  window.GOVUKFrontend.initAll({
+    accordion: {
+      rememberExpanded: false
+    },
+    characterCount: {
+      threshold: 90
+    }
+  })
+</script>
+```
+
+## Adding support for HTML `data-*` attributes
+
+For convenience, we allow for configuration options to be passed through HTML `data-*` attributes too.
+
+`data-*` attributes can be found in JavaScript by looking at an element's `dataset` property. Browsers will helpfully convert the attribute names from HTML's kebab-case to more JavaScript-friendly camelCase when working with `dataset`. See ['Naming configuration options'](#naming-configuration-options) for caveats.
+
+As we expect configuration-related `data-*` attributes to always be on the component's root element (the same element with the `data-module` attribute), we can access them all quickly using `$module.dataset`.
+
+Let's update our `mergeConfigs` call from earlier to include `$module.dataset` as the highest priority.
+
+```javascript
+import { mergeConfigs } from '../../common/index.mjs'
+import { normaliseDataset } from '../../common/normalise-dataset.mjs'
+
+this.config = mergeConfigs(
+  defaultConfig,
+  config || {},
+  normaliseDataset($module.dataset)
+)
+```
+
+Here, we pass it through our `normaliseDataset` function. This is because attribute values in dataset are always interpreted as being strings. `normaliseDataset` runs a few simple checks to convert values back to numbers or booleans where appropriate.
```

Comment:
```suggestion
Here, we pass `mergeConfigs` through our `normaliseDataset` function. This is because attribute values in dataset are always interpreted as strings. `normaliseDataset` runs a few simple checks to convert values back to numbers or booleans where appropriate.
```

Reply:
This isn't quite accurate. We're passing the value of `$module.dataset` through the `normaliseDataset` function, not the value or result of `mergeConfigs`. 

The code example does this inside the `mergeConfigs` function essentially for brevity, but it doesn't strictly have to be nested in this way.

Reply:
ok, so if we change 'it' to 'the value of `$module.dataset`', do we need to add further explanation or is that sufficient by itself?

Reply:
I think that should be sufficient, yeah

Reply:
Fab, thank you